src/Factory.php

```diff
@@ -440,4 +479,56 @@ private function isPersisting(): bool
             return false;
         }
     }
+
+    private function resetDoctrineEvents(): void
+    {
+        $this->doctrineEvents = true;
+        $this->doctrineEventTypeName = null;
+
+        /** @var EntityManagerInterface $manager */
+        $manager = self::configuration()->objectManagerFor($this->class);
+        /** @var ContainerAwareEventManager $eventManager */
+        $eventManager = $manager->getConnection()->getEventManager();
+
+        foreach ($this->disabledDoctrineEvents as $type => $listenersByType) {
+            foreach ($listenersByType as $listener) {
+                if ($listener instanceof EventSubscriber) {
+                    $eventManager->addEventSubscriber($listener);
+
+                    continue;
+                }
+
+                $eventManager->addEventListener($type, $listener);
+            }
+        }
+    }
+
+    private function disableDoctrineEvents(): void
+    {
+        /** @var EntityManagerInterface $manager */
+        $manager = self::configuration()->objectManagerFor($this->class);
+        /** @var ContainerAwareEventManager $eventManager */
+        $eventManager = $manager->getConnection()->getEventManager();
+
+        $this->disabledDoctrineEvents = $manager->getConnection()->getEventManager()->getListeners($this->doctrineEventTypeName);
+        foreach ($this->disabledDoctrineEvents as $type => $listenersByType) {
+            foreach ($listenersByType as $name => $listener) {
+                if ($listener instanceof EventSubscriber) {
+                    $manager->getConnection()->getEventManager()->removeEventSubscriber($listener);
+
+                    continue;
+                }
+
+                if (false === \mb_strpos($name, '_service_')) {
```

Comment:
Yeah actually I'm not really satisfied by this code, but I don't see how to do better.  
It is well tested, if this keyword change, it will break my test.

Reply:
I guess this is all we can do.